src/Resources/js/views/mealIndex.js

```diff
@@ -0,0 +1,89 @@
+export default function MealIndexView() {
+    this.updateSlots();
+    setInterval(this.updateSlots, 5000);
+
+    this.initEvents();
+};
+
+MealIndexView.prototype.initEvents = function () {
+    // set handler for slot change event
+    $('.meals-list .meal .slot-selector').change(this.handleChangeSlot);
+    $('.meals-list .meal .participation-checkbox').change(this.handleParticipationUpdate);
+}
+
+MealIndexView.prototype.handleChangeSlot = function (event) {
+    const $slotSelector = $(event.target);
+    const $mealContainer = $slotSelector.closest('.meal');
+    const mealIsBooked = $mealContainer.find('input[type="checkbox"]').is(':checked');
+
+    if (mealIsBooked) {
+        const $mealDate = $mealContainer.data('date');
+        const slot = $slotSelector.val();
+        $.ajax({
+            method: 'POST',
+            url: '/menu/meal/'+$mealDate+'/update-slot',
+            data: { 'slot': slot },
+            dataType: 'json',
+            error: function () {
+                alert('An unknown error occurred');
+            }
+        });
+    }
+}
+
+MealIndexView.prototype.handleParticipationUpdate = function (event) {
+    const $dishCheckbox = $(event.target);
+    const $mealContainer = $dishCheckbox.closest('.meal');
+    const bookedMealCount = $mealContainer.find('checkbox.participation-checkbox').length
+
+    // do nothing if user ia joining a meal
+    if ($dishCheckbox.is(':checked')) {
+        return;
+    }
+
+    // reset slot selector if user cancelled all booked meals
+    if (1 > bookedMealCount) {
+        let $slotSelector = $mealContainer.find('.slot-selector');
+        $slotSelector.val('');
+    }
+}
+
+MealIndexView.prototype.updateSlots = function () {
+    $.ajax({
+        'url': '/participation/slots-status',
+        dataType: 'json',
+        'success': function (data) {
+            $.each(data, function (k, v) {
+                const slotSelectorId = 'day-'+v.date.replaceAll('-', '')+'-slots';
+
+                let $slotSelector = $('#'+slotSelectorId);
+                if ($slotSelector.length < 1) {
+                    return;
+                }
+
+                let $slotOption = $slotSelector.find('option[value="'+v.slot+'"]');
+
+                const slotLimit = $slotOption.data('limit');
+                if (slotLimit > 0) {
+                    const slotTitle = $slotOption.data('title');
+                    const slotText = slotTitle + ' (' + v.booked+'/'+slotLimit + ')';
+                    $slotOption.text(slotText);
+                    // disable slot-option if no. of booked slots reach the slot limit
+                    if (slotLimit <= v.booked) {
```

Comment:
Same here. Don't get it :-)

You could also write it shorter here: `$slotOption.prop('disabled', slotLimit <= v.booked));`